File: manprobackend/app/Http/Controllers/FoundationProfileController.php
<?php

namespace App\Http\Controllers;

use App\Models\FoundationProfile;

class FoundationProfileController extends Controller
{
    public function index()
    {
        $profile = FoundationProfile::first();
        
        return response()->json([
            'status' => 'success',
            'data' => [
                'description' => $profile ? $profile->description : ''
            ]
        ]);
    }
} 

File: manproapp/lib/features/bagian_utama/controllers/foundation_profile_controller.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../../../utils/constants/api_constants.dart';
import '../Models/foundation_profile.dart';

class FoundationProfileController {
  Future<FoundationProfile> getProfile() async {
    try {
      final response = await http.get(Uri.parse('${url}foundation-profile'));

      if (response.statusCode == 200) {
        final Map<String, dynamic> data = json.decode(response.body);
        if (data['status'] == 'success' && data['data'] != null) {
          return FoundationProfile.fromJson(data['data']);
        }
      }
      throw Exception('Failed to load foundation profile');
    } catch (e) {
      throw Exception('Error: $e');
    }
  }
} 